packages/structure/src/model/RWRoute.ts

```diff
@@ -38,6 +38,16 @@ export class RWRoute extends BaseNode {
     return LocationLike_toLocation(this.jsxNode)
   }
 
+  @lazy() get hasInvalidPrivateRoute() {
+    // Return `true` when a notfound route is nested within a Private Tag
+    if (typeof this.jsxNode.getAttribute('notfound') !== 'undefined') {
+      return this.jsxNode.compilerNode.parent
+        .getFullText()
+        .startsWith('<Private')
```

Comment:
This was the best way I could figure out how to access the `tagName` for `parent` as the methods available to `jsxNode` aren't available for the `parent` node but I am open to suggestions re: different approaches to determining if the parent node is a Private route.

Reply:
Hi Monica,

I think what you're doing is ok for now. Traversing JSX is tricky.

Please extract this to a separate getter `@lazy() get isPrivate(): boolean {...}`. It is important to keep all these heuristics consistent. We don't want some algorithms to use your strategy, and then someone else reinventing the wheel.

FYI: the ts-morph way of doing this would probably be too verbose, which is why I think your strategy is ok for now.

```ts
    const tagText = this.jsxNode
      .getParentIfKind(tsm.SyntaxKind.JsxElement)
      ?.getOpeningElement()
      ?.getTagNameNode()
      ?.getText()
```

Reply:
Thanks for the feedback. I refactored the code based on your feedback and it worked both with and without a comment infront of the <Private> tag from my manual testing.